e2e/cypress/integration/enterprise/system_console/channel_moderation_spec.js

```diff
@@ -35,24 +59,876 @@ const enableAllChannelModeratedPermissions = () => {
+    });
+};
+
+// # Checks to see if we did not get a system message warning after using @all/@here/@channel
+const postChannelMentionsAndVerifySystemMessageNotExist = () => {
+    cy.findByTestId('post_textbox').clear().type('@all{enter}{enter}{enter}');
+
+    // # Get last post message text
+    cy.getLastPostId().then((postId) => {
+        // * Assert that the last message posted is NOT a system message informing us we are not allowed to use channel mentions
+        cy.get(`#postMessageText_${postId}`).should('not.have.text', 'Channel notifications are disabled in aut-8. The @all did not trigger any notifications.');
+    });
+
+    cy.findByTestId('post_textbox').clear().type('@here{enter}{enter}{enter}');
+
+    // # Get last post message text
+    cy.getLastPostId().then((postId) => {
+        // * Assert that the last message posted is NOT a system message informing us we are not allowed to use channel mentions
+        cy.get(`#postMessageText_${postId}`).should('not.have.text', 'Channel notifications are disabled in aut-8. The @here did not trigger any notifications.');
+    });
+
+    cy.findByTestId('post_textbox').clear().type('@channel{enter}{enter}{enter}');
+
+    // # Get last post message text
+    cy.getLastPostId().then((postId) => {
+        // * Assert that the last message posted is NOT a system message informing us we are not allowed to use channel mentions
+        cy.get(`#postMessageText_${postId}`).should('not.have.text', 'Channel notifications are disabled in aut-8. The @channel did not trigger any notifications.');
+    });
+};
+
 describe('Channel Moderation Test', () => {
     before(() => {
         // * Check if server has license
         cy.requireLicense();
+    });
 
+    beforeEach(() => {
         // Reset permissions in system scheme to defaults.
+        resetSystemSchemePermissionsToDefault();
+
+        // # Delete all Team Override Schemes
         cy.apiLogin('sysadmin');
-        cy.visit('/admin_console/user_management/permissions/system_scheme');
-        cy.findByTestId('resetPermissionsToDefault').click();
-        cy.get('#confirmModalButton').click();
-        cy.get('#saveSetting').click();
-        waitUntilConfigSave();
+        cy.visit('/admin_console/user_management/permissions');
+        cy.get('#team-override-schemes-list').then((elements) => {
+            for (let i = 1; i < elements[0].children.length && elements[0].children[1].className !== 'no-team-schemes'; i++) {
+                elements[0].children[i].children[1].children[1].lastChild.click();
+                cy.get('#confirmModalButton').click({force: true});
+            }
+        });
+
+        // # Reset Autem Channel Moderation settings to default (everything on)
+        visitAutemChannelConfigPage();
+        enableAllChannelModeratedPermissions();
+        saveConfig();
```

Comment:
Just a suggestion: 
Since these steps are being called before each test, instead of doing this in the UI, can we make an API request since it will be much faster & less prone to errors.

Reply:
True that! thanks for pointing this out.